src/pudl/transform/epacems.py

```diff
@@ -18,16 +18,79 @@
 ###############################################################################
 
 
-def fix_up_dates(df: pd.DataFrame, plant_utc_offset: pd.DataFrame) -> pd.DataFrame:
-    """Fix the dates for the CEMS data.
+def harmonize_eia_epa_orispl(
+    df: pd.DataFrame,
+    pudl_engine: sa.engine.Engine,
+) -> pd.DataFrame:
+    """Harmonize the ORISPL code to match the EIA data.
+
+    The EIA plant IDs and CEMS ORISPL codes almost match, but not quite. EPA has
+    compiled a crosswalk that maps one set of IDs to the other. The crosswalk is
+    integrated into the PUDL db.
+
+    EIA IDs are more correct so use the crosswalk to fix any erronious EPA IDs and get
+    rid of that column to avoid confusion.
+
+    https://github.com/USEPA/camd-eia-crosswalk
+
+    Note that this transformation needs to be run *before* convert_to_utc, because
+    convert_to_utc uses the plant ID to look up timezones.
+
+    Args:
+        pudl_engine: SQLAlchemy connection engine for connecting to an existing PUDL DB.
+            This is used to access the crosswalk file for conversion. The crosswalk must
+            be processed prior to running this function or it won't work.
+        df: A CEMS hourly dataframe for one year-month-state.
+
+    Returns:
+        The same data, with the ORISPL plant codes corrected to match the EIA plant IDs.
+
+    """
+    # Already ran a test to make sure this works. When you group the crosswalk by
+    # plant_id_epa and emissions_unit_id_epa then calculate .nunique() for plant_id_eia,
+    # none of the values are greater than one meaning that this drop/merge is ok. Might
+    # want to make that an official test somewhere.
+
+    crosswalk_df = pd.read_sql(
+        "epacamd_eia_crosswalk",
+        con=pudl_engine,
+        columns=["plant_id_eia", "plant_id_epa", "emissions_unit_id_epa"],
+    ).drop_duplicates()
```

Comment:
Can we assume that the pudl db will have this table when this function is used? Is this managed via the etl module's sequencing of the data sets? I know there was already this coupling of cems and eia, so I assume this is not a new issue..

Reply:
This check is in the `etl_epacems()` function in the `etl` module

```
    # Verify that we have a PUDL DB with crosswalk data
    if "epacamd_eia" not in inspector.get_table_names():
        raise RuntimeError(
            "No EPACAMD-EIA Crosswalk available in the PUDL DB! Have you run the ETL? "
            f"Trying to access PUDL DB: {pudl_engine}"
        )
```